packages/gpac/build.sh

```diff
@@ -5,29 +5,20 @@ TERMUX_PKG_LICENSE="LGPL-2.1"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION="2.4.0"
 TERMUX_PKG_REVISION=1
-TERMUX_PKG_SRCURL=https://github.com/gpac/gpac/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=99c8c994d5364b963d18eff24af2576b38d38b3460df27d451248982ea16157a
+TERMUX_PKG_SRCURL=(
+	https://github.com/gpac/gpac/archive/refs/tags/v"${TERMUX_PKG_VERSION}".tar.gz
+	https://github.com/gpac/gpac/commit/18863aa2176e423dae2a6d7e39ff6ed6a37b2b78.patch
+)
+TERMUX_PKG_SHA256=(
+	99c8c994d5364b963d18eff24af2576b38d38b3460df27d451248982ea16157a
+	3a4e10a031bc081a402bfd24889f85fcbf99d4fd36a9086aedda546445037bec
+)
 TERMUX_PKG_DEPENDS="ffmpeg, freetype, liba52, libjpeg-turbo, liblzma, libmad, libnghttp2, libogg, libpng, libtheora, libvorbis, openjpeg, openssl, pulseaudio, xvidcore, zlib"
 TERMUX_PKG_EXTRA_MAKE_ARGS="STRIP=:"
 TERMUX_PKG_EXTRA_CONFIGURE_ARGS="--disable-x11"
 
 termux_step_pre_configure() {
-	declare -a _commits=(
-	18863aa2
-	)
-
-	declare -a _checksums=(
-	3a4e10a031bc081a402bfd24889f85fcbf99d4fd36a9086aedda546445037bec
-	)
-
-	for i in "${!_commits[@]}"; do
-		PATCHFILE="${TERMUX_PKG_CACHEDIR}/gpac_patch_${_commits[i]}.patch"
-		termux_download \
-			"https://github.com/gpac/gpac/commit/${_commits[i]}.patch" \
-			"$PATCHFILE" \
-			"${_checksums[i]}"
-		patch -p1 -i "$PATCHFILE"
-	done
+	patch -p1 -i "${TERMUX_PKG_CACHEDIR}"/18863aa2176e423dae2a6d7e39ff6ed6a37b2b78.patch
```

Comment:
Would it make sense to try to make the `.patch` files automatically apply without needing to manually call `patch` within `build.sh` (like patches in the `TERMUX_PKG_BUILDER_DIR`), or is there a downside to that which isn't immediately obvious?

https://github.com/termux/termux-packages/blob/042a11491b71a0fc1e9dbe1e1cadb7994014ce4d/scripts/build/termux_step_patch_package.sh#L20-L30

Reply:
Some patches can be download to revert them, e.g. HandBrake. So, I though that adding the patch command manually would make it easier.

Reply:
in that case I guess that the `.diff` form of the filename can be used (they are interchangeable in GitHub URLs) but that makes sense, it's an abstraction that doesn't hold for everything (just like the `TERMUX_PKG_BUILDER_DIR` patches, some of which are named `.diff`)

Reply:
We could add an overridable function for handling upstream patches.
That way we can have a default, and let it be overwritten if necessary.